kroxylicious-runtime/src/test/java/io/kroxylicious/proxy/internal/KafkaProxyFrontendHandlerTest.java

```diff
@@ -379,11 +409,37 @@ void expectedFlow(boolean sslConfigured,
         var handler = handler(filter, dp, endpointBinding);
         initialiseInboundChannel(handler);
 
+        if (clientAuthConfigured) {
+            try {
+                when(sslSession.getPeerPrincipal()).thenReturn(mockPrincipal);
+            }
+            catch (SSLPeerUnverifiedException e) {
+                throw new RuntimeException(e);
+            }
+        }
+        else {
+            try {
+                when(sslSession.getPeerPrincipal()).thenThrow(new SSLPeerUnverifiedException("No peer certificate"));
+            }
+            catch (SSLPeerUnverifiedException e) {
+                throw new RuntimeException(e);
+            }
+        }
+
         if (sslConfigured) {
             // Simulate the SSL handler
             inboundChannel.pipeline().fireUserEventTriggered(new SniCompletionEvent(SNI_HOSTNAME));
+            inboundChannel.pipeline().fireUserEventTriggered(SslHandshakeCompletionEvent.SUCCESS);
         }
 
+        if (sslConfigured) {
+            if (clientAuthConfigured) {
+                assertEquals(CLIENT_PRINCIPAL, handler.getDownstreamCertificatePrincipal().getName());
+            }
+            else {
+                assertEquals("ANONYMOUS", handler.getDownstreamCertificatePrincipal().getName());
```

Comment:
As a project we generally use [AssertJ](https://assertj.github.io/doc/) so we'd write this as 
```suggestion
                assertThat(handler.getDownstreamCertificatePrincipal()).isEqualTo(KafkaPrincipal.ANONYMOUS);
```

Though I appreciate `assertEquals` is comonly used in this test.